integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/localdictionary/LocalDictionarySupportAlterTableTest.scala

```diff
@@ -0,0 +1,1183 @@
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,st struct<s_id:int,
+        | s_name:int,s_city:array<int>>, dcity array<int>)
+        | STORED BY 'org.apache.carbondata.format'
+        | tblproperties('local_dictionary_exclude'='name',
+        | 'local_dictionary_enable'='true')
+      """.
+        stripMargin)
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_exclude'='dcity')")
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_include'='st')")
+    }
+  }
+
+  test("test alter unset for local dictionary disable")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,add string)
+        | STORED BY 'carbondata' tblproperties('local_dictionary_enable'='false')
+      """.stripMargin)
+
+    val descLoc1 = sql("describe formatted local1").collect
+    descLoc1.find(_.get(0).toString.contains("Local Dictionary Enable")) match {
+      case Some(row) => assert(row.get(1).toString.contains("false"))
+    }
+    sql("alter table local1 unset tblproperties('local_dictionary_enable')")
+    val descLoc2 = sql("describe formatted local1").collect
+    descLoc2.find(_.get(0).toString.contains("Local Dictionary Enable")) match {
+      case Some(row) => assert(row.get(1).toString.contains("true"))
+    }
+  }
+
+  test("test alter unset for local dictionary enable local dict include")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,add string)
+        | STORED BY 'carbondata' tblproperties('local_dictionary_include'='city')
+      """.stripMargin)
+
+    val descLoc1 = sql("describe formatted local1").collect
+    descLoc1.find(_.get(0).toString.contains("Local Dictionary Include")) match {
+      case Some(row) => assert(row.get(1).toString.contains("city"))
+    }
+    sql("alter table local1 unset tblproperties('local_dictionary_include')")
+    val descLoc2 = sql("describe formatted local1").collect
+    descLoc2.find(_.get(0).toString.contains("Local Dictionary Include")) match {
+      case Some(row) => assert(row.get(1).toString.contains("name,city"))
+    }
+  }
+
+  test("test alter unset for local dictionary enable local dict exclude")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,add string)
+        | STORED BY 'carbondata' tblproperties('local_dictionary_include'='city',
+        | 'local_dictionary_exclude'='name')
+      """.stripMargin)
+
+    val descLoc1 = sql("describe formatted local1").collect
+    descLoc1.find(_.get(0).toString.contains("Local Dictionary Include")) match {
+      case Some(row) => assert(row.get(1).toString.contains("city"))
+    }
+    sql("alter table local1 unset tblproperties('local_dictionary_include')")
```

Comment:
unset 'local_dictionary_exclude' instead of 'local_dictionary_include' as the name suggests

Reply:
handled in #2450